modules/nw-ovn-kubernetes-migration.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+
+[id="nw-ovn-kubernetes-migration_{context}"]
+= Migrating to the OVN-Kubernetes network provider
+
+As a cluster administrator, you can upgrade the Pod network provider for your cluster to the OVN-Kubernetes Pod network provider.
+During the migration, you must reboot every node in your cluster.
+
+[IMPORTANT]
+====
+While performing the migration, your cluster is unavailable and workloads may be interrupted.
+Perform the migration only when an interruption in service is acceptable.
+====
+
+.Prerequisites
+
+* Install the OpenShift Command-line Interface (CLI), commonly known as `oc`.
+* Access to the cluster as a user with the `cluster-admin` role.
+* A cluster installed on bare metal infrastructure configured with the OpenShift SDN Pod network provider.
+* The cluster is in a known good state, without any errors.
+
+.Procedure
+
+. To backup the configuration for the cluster network, enter the following command:
++
+----
+$ oc get Network.config.openshift.io cluster -o yaml
+----
+
+. To enable the migration, set an annotation on the Cluster Network Operator configuration object by entering the following command:
++
+----
+$ oc annotate Network.operator.openshift.io cluster \
+  'networkoperator.openshift.io/network-migration'=""
+----
+
+. To change the network provider, enter the following command:
++
+----
+$ oc patch Network.config.openshift.io cluster \
+  --type='merge' --patch '{ "spec": { "networkType": "OVNKubernetes" } }'
+----
+
+. To confirm the migration disabled the OpenShift SDN network provider, enter the following command. It might take several moments for all the OpenShift SDN Pods to stop.
++
+----
+$ oc get pod -n openshift-sdn --watch
+----
+
+. To complete the migration, reboot each node in your cluster. For example, the following bash script reboots each node in your cluster. The script assumes that you can connect to each host by using `ssh` and that you have configured `sudo` to not prompt for a password.
++
+[source,bash]
+----
+#!/bin/bash
+
+for ip in $(oc get nodes  -o jsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+do
+   echo "reboot node $ip"
+   ssh -o StrictHostKeyChecking=no core@$ip sudo shutdown -r -t 3
+done
+----
+
+. Confirm that the migration succeeded:
+
+.. To confirm that the Pod network provider is OVN-Kubernetes, enter the following command:
++
+----
+$ oc describe clusteroperators/network
+----
+
+.. To confirm that the cluster nodes are in the `Ready` state, enter the following command:
++
+----
+$ oc get nodes
+----
++
+If a node is stuck in the `NotReady` state, reboot the node again.
+
+. To remove the migration annotation from the Cluster Network Operator configuration object, enter the following command:
```

Comment:
I think we should say. If after the migration, the cluster is back to a normal state and a rollback is not needed. You shall 1) remove the `openshift-sdn` manually. 2) remove the migration annotation.